Sources/SwiftBasicFormat/BasicFormat.swift

```diff
@@ -186,12 +186,12 @@ open class BasicFormat: SyntaxRewriter {
     }
 
     switch (first?.tokenKind, second?.tokenKind) {
-
     case (.multilineStringQuote, .backslash),  // string interpolation segment inside a multi-line string literal
       (.multilineStringQuote, .multilineStringQuote),  // empty multi-line string literal
       (.multilineStringQuote, .stringSegment),  // segment starting a multi-line string literal
       (.stringSegment, .multilineStringQuote),  // ending a multi-line string literal that has a string interpolation segment at its end
       (.rightParen, .multilineStringQuote),  // ending a multi-line string literal that has a string interpolation segment at its end
+      (.poundEndifKeyword, _),
```

Comment:
Do we also need `.poundSourceLocationKeyword` here?

Reply:
Don't this its needed. 

The reason I added that extra one is because 
```swift
#if hasAttribute(foo)
      @foo
      #endif
      struct MyStruct {}
```

becomes this without 
```swift
#if hasAttribute(foo)
@foo
#endif struct MyStruct {
}
```

Reply:
Oh, and `#sourceLocation` always has arguments, so this doesn’t apply. Makes sense 👍🏽